test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -437,4 +437,19 @@ spec = Test.group "Dataflow Warnings" <|
         result_non_tail . should_equal 6
         Warning.get_all result_non_tail . map .value . should_equal ["Foo"]
 
```

Comment:
I'm wondering if it's worth making this a separate test case:
```suggestion
Test.specify "should not break TCO when warnings are attached to arguments" <|
```

Reply:
I did wonder about it but then I was like :shrug:  :)

Reply:
IMO worth adding as it clarifies what this is testing

Reply:
sure, np